src/transformers/__tests__/pivot.test.ts

```diff
@@ -0,0 +1,161 @@
+import { uncheckedPivotLonger, uncheckedPivotWider } from "../pivot";
+import {
+  CENSUS_DATASET,
+  cloneDataSet,
+  DATASET_A,
+  EMPTY_RECORDS,
+  GRADES_DATASET_LONGER,
+  GRADES_DATASET_WIDER,
+  TYPES_DATASET,
+} from "./data";
+
+describe("pivot wider", () => {
+  it("errors on dataset with more than 1 collection", () => {
+    expect(() => uncheckedPivotWider(DATASET_A, "A", "B")).toThrow(
+      "single-collection"
+    );
+  });
+  it("errors on bad namesFrom attribute", () => {
+    expect(() =>
+      uncheckedPivotWider(DATASET_A, "AttributeThatDoesNotExist", "A")
+    ).toThrow("AttributeThatDoesNotExist");
+  });
+  it("errors on bad valuesFrom attribute", () => {
+    expect(() =>
+      uncheckedPivotWider(DATASET_A, "A", "AttributeThatDoesNotExist")
+    ).toThrow("AttributeThatDoesNotExist");
+  });
+  it("errors on empty dataset", () => {
+    const EMPTY_RECORDS_SINGLE_COLLECTION = cloneDataSet(EMPTY_RECORDS);
+    EMPTY_RECORDS_SINGLE_COLLECTION.collections = [
+      EMPTY_RECORDS_SINGLE_COLLECTION.collections[0],
+    ];
+    const output = {
+      collections: [
+        {
+          attrs: [{ formula: undefined, name: "C" }],
+          name: "Collection A",
+          parent: undefined,
+        },
+      ],
+      records: [],
+    };
+    expect(
+      uncheckedPivotWider(EMPTY_RECORDS_SINGLE_COLLECTION, "A", "B")
+    ).toEqual(output);
+  });
+
+  it("errors on non-displayable value in  valuesFrom attribute", () => {
+    expect(() =>
+      uncheckedPivotWider(TYPES_DATASET, "Boundary", "Number")
+    ).toThrow("Cannot use a boundary");
+  });
+  it("allows non-displayable value in  namesFrom attribute", () => {
+    expect(() =>
+      uncheckedPivotWider(TYPES_DATASET, "Number", "Boundary")
+    ).not.toThrow();
+  });
```

Comment:
These tests look good, but I think the attribute names (in the descriptions) are swapped--pivot wider errors on non-displayable values in `namesFrom` because they are used as attribute names, but allows them in `valuesFrom` since they stay values.